sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs

```diff
@@ -176,630 +123,292 @@ public virtual Pageable<string> GetMetricDimensionValues(string metricId, string
+                queryOptions.StartTime = filter.StartsOn;
+                queryOptions.TimeMode = filter.TimeMode;
             }
 
-            async Task<Page<MetricSeriesDefinition>> FirstPageFunc(int? pageSizeHint)
+            int? skip = options?.Skip;
+            int? maxPageSize = options?.MaxPageSize;
+            RequestContext context = new RequestContext()
             {
-                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(GetMetricSeriesDefinitions)}");
-                scope.Start();
+                CancellationToken = cancellationToken,
+            };
 
-                try
-                {
-                    Response<MetricSeriesList> response = await _serviceRestClient.GetMetricSeriesAsync(metricGuid, queryOptions, skip, maxPageSize, cancellationToken).ConfigureAwait(false);
-                    return Page.FromValues(response.Value.Value, response.Value.NextLink, response.GetRawResponse());
-                }
-                catch (Exception e)
-                {
-                    scope.Failed(e);
-                    throw;
-                }
-            }
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(GetAllFeedback)}");
+            scope.Start();
 
-            async Task<Page<MetricSeriesDefinition>> NextPageFunc(string nextLink, int? pageSizeHint)
+            try
             {
-                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(GetMetricSeriesDefinitions)}");
-                scope.Start();
-
-                try
-                {
-                    Response<MetricSeriesList> response = await _serviceRestClient.GetMetricSeriesNextAsync(nextLink, queryOptions, cancellationToken).ConfigureAwait(false);
-                    return Page.FromValues(response.Value.Value, response.Value.NextLink, response.GetRawResponse());
-                }
-                catch (Exception e)
-                {
-                    scope.Failed(e);
-                    throw;
-                }
+                RequestContent content = MetricFeedbackFilter.ToRequestContent(queryOptions);
+                AsyncPageable<BinaryData> pageableBindaryData = GetMetricFeedbacksAsync(content, skip, maxPageSize, context);
+                return PageableHelpers.Select(pageableBindaryData, response => MetricFeedbackList.FromResponse(response).Value);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
             }
-
-            return PageableHelpers.CreateAsyncEnumerable(FirstPageFunc, NextPageFunc);
         }
 
         /// <summary>
-        /// Gets a collection of items describing the time series of a specified <see cref="DataFeedMetric"/>.
+        /// Gets a collection of <see cref="MetricFeedback"/> related to the given metric.
```

Comment:
Curious: If this method is manually written, why not preserve the hand-written comments from the HLC library?

Reply:
Ah, looks like it may just be a bad diff.